modules/host/negotiate.go

```diff
@@ -143,6 +143,9 @@ func (h *Host) addCollateral(txn consensus.Transaction, terms modules.ContractTe
 	if err != nil {
 		return
 	}
+	if collateral.Cmp(consensus.NewCurrency64(0)) == 0 {
```

Comment:
can use `consensus.ZeroCurrency` for this. I really ought to bring back `IsZero` though.